src/apps/datacopy.c

```diff
@@ -655,7 +655,10 @@ transfer_data(const BCPPARAMDATA * params, DBPROCESS * dbsrc, DBPROCESS * dbdest
 	}
 
 	srcdata = (MIGCOLDATA *) calloc(sizeof(MIGCOLDATA), src_numcols);
-
+	if(srcdata == NULL){
```

Comment:
About syntax, code uses a space after if and before bracket, also prefer `!ptr` instead of `ptr == NULL`, so

```c
if (!srcdata) {
```

I can do the change if you prefer